### PR TITLE
fix(meta): fix trivial loop bug

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by MrCroxx — APPROVED** [review]

File: src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs [hunk 1]
```diff
@@ -238,7 +238,13 @@ impl NonOverlapSubLevelPicker {
 
             // more than 1 sub_level
             if ret.total_file_count > 1 && ret.total_file_size >= self.max_compaction_bytes
```
Comment (Li0k) [inline src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs:240]: for this picker, "ret.total_file_count > 1" is enough ? It means choosing more the 1 sub_level. However, both LGTM

**Review by Li0k — APPROVED** [review]
Rest LGTM

**Comment by hzxa21** [comment]
Open a PR for main as well?

File: src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs:240]: Why?
ret.total_file_count > 1 means that there may be only one sub-level with more than 1 files.

Comment (Li0k) [reply src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs:240]: We choose the basic sst in the input level (at least one), So, when sst in ret is greater than 1, it means we select other sub-level

**Comment by lmatz** [comment]
does 1.8 need this fix?

**Comment by Little-Wallace** [comment]
> Open a PR for main as well?

I submit a PR in https://github.com/risingwavelabs/risingwave/pull/16069

**Comment by Little-Wallace** [comment]
> does 1.8 need this fix?

I think 1.8 also need, but this bug is not easy to meet in 1.8 because 1.8 has fixed the small files bug.

**Review by Li0k — APPROVED** [review]
LGTM

**Pull request merged**